tools/debug/src/debug_ui.rs

```diff
@@ -0,0 +1,100 @@
+//! Debugging user interface for development
+use crate::*;
+
+/// Generate debug labels for tile positions
+pub fn generate_debug_labels(
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    tilemap_q: Query<(&Transform, &TilemapType, &TilemapGridSize)>, // , With<TerrainTilemap>
+    tile_q: Query<&TilePos>,
+) {
+    let font = asset_server.load("fonts/FiraSans-Bold.ttf");
+    let text_style = TextStyle {
+        font,
+        font_size: 15.0,
+        color: Color::BLACK,
+    };
+    let text_alignment = TextAlignment::CENTER;
+
+    for q in &tilemap_q {
+        let (tilemap_transform, map_type, grid_size) = q;
+
+        let label_bundles: Vec<Text2dBundle> = tile_q
+            .iter()
+            .map(|tile_pos| {
+                let tile_pos_transform = Transform::from_translation(
+                    tile_pos.center_in_world(grid_size, map_type).extend(1.0),
+                );
+                let transform = *tilemap_transform * tile_pos_transform;
+                Text2dBundle {
+                    text: Text::from_section(
+                        format!("{}, {}", tile_pos.x, tile_pos.y),
+                        text_style.clone(),
+                    )
+                    .with_alignment(text_alignment),
+                    transform,
+                    ..Default::default()
+                }
+            })
+            .collect();
+        commands.spawn_batch(label_bundles);
+    }
+}
+
+// Modified text_debug example from the Bevy UI examples (https://github.com/bevyengine/bevy/blob/main/examples/ui/text_debug.rs)
+/// Tag for changing text ui elements
+#[derive(Component)]
+pub struct TextChanges;
```

Comment:
This should be called `FpsText`?

Reply:
Is `ChangingText` okay, that way we can use it for other changing text elements and make our `change_infotext` query equally generic?

Reply:
@wallamide well, the design sense I have ended up developing (which might not be right) is that you'll often want to end up querying for specific things

so for example, in the `change_infotext` system, you'll have:

```
fps_text: Query<&mut Text, With<FpsText>>,
other_text: Query<&mut Text, With<OtherText>>,

...

for text in fps_text.iter_mut() { ... do fps specific stuff }

for text in other_text.iter_mut() { ... do other specific stuff }

```

Essentially the idea is to make the query as specific as possible, so that you are only iterating over relevant things, rather than iterating over something more general, and then (somehow? how?) branching inside the for loop:

```
for text in changing_text.iter_mut() {
    if text is fps {... do fps stuff ...}
    if text is other {...do other...}
}
```

@alice-i-cecile (is what I am saying right?)

Reply:
Oh that makes perfect sense. I'll go ahead and change it to `FpsText`!